packages/@aws-cdk/aws-appsync/lib/private.ts

```diff
@@ -203,8 +208,8 @@ function generateInterfaces(interfaceTypes?: InterfaceType[]): string {
  * @param directives the directives of a given type
  * @param delimiter the separator betweeen directives (by default we will add a space)
  */
-function generateDirectives(directives?: Directive[], delimiter?: string): string {
+function generateDirectives(directives?: Directive[], delimiter?: string, modes?: AuthorizationType[]): string {
   if (!directives || directives.length === 0) return '';
```

Comment:
unrelated to changes in this PR, but since you're in here can you put this if condition logic into braces?